rotkehlchen/chain/ethereum/modules/liquity/decoder.py

```diff
@@ -131,17 +138,72 @@ def _decode_stability_pool_event(
                     tx_log.topics[0] == STABILITY_POOL_LQTY_PAID
                 )
             ):
+                event.event_type = HistoryEventType.STAKING
                 event.event_subtype = HistoryEventSubType.REWARD
                 event.counterparty = CPT_LIQUITY
                 event.notes = f"Collect {event.balance.amount} {event.asset.resolve_to_crypto_asset().symbol} from liquity's stability pool"  # noqa: E501
         return None, []
 
+    def _decode_lqty_staking_deposits(
+            self,
+            tx_log: EthereumTxReceiptLog,
+            transaction: EvmTransaction,  # pylint: disable=unused-argument
+            decoded_events: List[HistoryBaseEntry],
+            all_logs: List[EthereumTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: List[ActionItem],  # pylint: disable=unused-argument
+    ) -> Tuple[Optional[HistoryBaseEntry], List[ActionItem]]:
+        if tx_log.topics[0] not in STAKING_LQTY_EVENTS:
+            return None, []
+
+        informational_event = None
+        for event in decoded_events:
+            if tx_log.topics[0] == STAKING_LQTY_CHANGE:
+                user = hex_or_bytes_to_address(tx_log.topics[1])
```

Comment:
Why to decode user address and lqty_amount on each iteration? I would say that this has to be done only once